spec/factories/announcements.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: announcements
+#
+#  id         :integer          not null, primary key
+#  subject    :text             not null
+#  details    :text             not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+FactoryBot.define do
+  factory :announcement do |a|
+    a.subject 'MyText'
+    a.details 'MyText'
+  end
+
+  factory :invalid_subject, class: Announcement do
```

Comment:
I don't think you need this? Just do `create(:announcement, subject: nil)`. Let's not clutter our factories haha